### PR TITLE
Detect unused coupled variables 

### LINKED ISSUE #22389: `coupledValue` without a component parameter should only work if the user specified exactly _one_ variable
## Reason
`addCoupledVar` let's the users add an arbitrary number of variables. Many object implicitly assume only one variable is passed in and fetch its value using `coupledValue` without the optional `comp` parameter. If a user specifies more than one variable in teh input file, the remaining variables get silently ignored.

## Design
Default `comp` to `libMesh::invalid_uint` and return the first variable for that index, iff only _one_ variable was specified. Otherwise throw an error.

## Impact
Better input file checking.

### PR DESCRIPTION
**Pull request by dschwen**
Closes #22389

### CODE REVIEW

File: modules/chemical_reactions/src/kernels/CoupledBEEquilibriumSub.C [hunk 1]
```diff
@@ -41,11 +41,11 @@ CoupledBEEquilibriumSub::CoupledBEEquilibriumSub(const InputParameters & paramet
     _sto_v(getParam<std::vector<Real>>("sto_v")),
     _gamma_u(coupledValue("gamma_u")),
     _gamma_u_old(coupledValueOld("gamma_u")),
-    _gamma_v(isCoupled("gamma_v")
+    _gamma_v(coupledComponents("gamma_v") > 1
```
Comment (dschwen) [inline modules/chemical_reactions/src/kernels/CoupledBEEquilibriumSub.C:44]: `isCoupled` will return false if none of the supplied input file values are variables (e.g. `gamma_v = '1 2 3 4'`)!

**Review by GiudGiud — DISMISSED** [review]

File: modules/chemical_reactions/src/kernels/CoupledBEEquilibriumSub.C (hunk 1 above)
Comment (YaqiWang) [reply modules/chemical_reactions/src/kernels/CoupledBEEquilibriumSub.C]: Care to improve the doxygen comment of this function in the header file?

File: framework/include/interfaces/Coupleable.h [hunk 2]
```diff
@@ -158,7 +158,7 @@ class Coupleable
    * @see Kernel::_u
    */
   virtual const VariableValue & coupledValue(const std::string & var_name,
-                                             unsigned int comp = 0) const;
+                                             unsigned int comp = libMesh::invalid_uint) const;
```
Comment (YaqiWang) [inline framework/include/interfaces/Coupleable.h:161]: Lots of other functions have the 0 as the default. Do we need to change those?

Comment (dschwen) [reply framework/include/interfaces/Coupleable.h:161]: Yeah, that should be investigated. `coupledValue` is the most important one IMO, and I wanted to see how this impacts our tests.

File: modules/chemical_reactions/src/kernels/CoupledBEEquilibriumSub.C (hunk 1 above)
Comment (dschwen) [reply modules/chemical_reactions/src/kernels/CoupledBEEquilibriumSub.C]: Done

**Comment by lindsayad** [comment]
what a sad last commit message 😆 😢

**Pull request closed without merging**